hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/flakeidgenerator/NewIdBatchMessageTask.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task.flakeidgenerator;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.FlakeIdGeneratorNewIdBatchCodec;
+import com.hazelcast.client.impl.protocol.task.AbstractMessageTask;
+import com.hazelcast.concurrent.flakeidgen.FlakeIdGeneratorProxy;
+import com.hazelcast.concurrent.flakeidgen.FlakeIdGeneratorService;
+import com.hazelcast.core.FlakeIdGenerator.IdBatch;
+import com.hazelcast.instance.Node;
+import com.hazelcast.nio.Connection;
+import com.hazelcast.security.permission.ActionConstants;
+import com.hazelcast.security.permission.FlakeIdGeneratorPermission;
+
+import java.security.Permission;
+
+public class NewIdBatchMessageTask
+        extends AbstractMessageTask<FlakeIdGeneratorNewIdBatchCodec.RequestParameters> {
```

Comment:
We have `AbstractCallableMessageTask` which seems more appropriate for this message task.